x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelConfig.java

```diff
@@ -780,6 +776,7 @@ public Builder validate(boolean forCreation) {
                     ESTIMATED_OPERATIONS.getPreferredName(),
                     validationException);
                 validationException = checkIllegalSetting(licenseLevel, LICENSE_LEVEL.getPreferredName(), validationException);
+                validationException = checkIllegalSetting(location, LOCATION.getPreferredName(), validationException);
```

Comment:
?

Reply:
Ok I see now, the location is always a `.ml-inference-XXX` index and we can figure out which from the model id. 

I'm not sure about the one index per model pattern. Could we use a shared index and alias `.ml-inference-my-model` to `.ml-inference-shared`

Reply:
@davidkyle the whole point of this thing was to allow pytorch models to have a unique index. That is why we have "location" to begin with.

Reply:
Discussing further, the main issue is having a unique index for every model. I have changed it to be a shared index for all "native" models. We will restrict its size right now for sanity, but this restriction may change in the future.